src/fingerprints.rs

```diff
@@ -3,15 +3,36 @@ use crate::git::*;
 use std::error::Error;
 use log::*;
 
-pub fn read_fingerprints<G: Git>(team_fingerprints_file: &str) -> Result<Vec<String>, Box<dyn Error>>
-{
-    trace!("Fetching team fingerprints");
-    let git = G::new()?;
-    let fingerprints_file = git.read_file(team_fingerprints_file)?;
-    let fingerprints: Vec<String> = fingerprints_file.split('\n')
-        .filter_map( |s| s.split(',').next())
-        .map (|s| s.replace(char::is_whitespace, ""))
-        .filter(|s| !s.is_empty())
-        .collect();
-    return Ok(fingerprints);
+pub struct Fingerprint {
+    pub id: String,
+    pub name: String,
+    pub email: String
+}
+
+impl Fingerprint {
+    pub fn read_fingerprints<G: Git>(git: &G, team_fingerprints_file: &str) -> Result<Vec<Fingerprint>, Box<dyn Error>>
```

Comment:
Perhaps fingerprints should be a hashmap indexed on email address?
Otherwise the complexity of later operations becomes quadratic when we're in an inner loop looking for people's fingerprint entries.